api/preprints/views.py

```diff
@@ -350,3 +354,72 @@ def get_object(self):
             return {'citation': citation, 'id': style}
 
         raise PermissionDenied if auth.user else NotAuthenticated
+
+
+class PreprintIdentifierList(JSONAPIBaseView, generics.ListAPIView, ODMFilterMixin, PreprintMixin):
+    """List of identifiers for a specified preprint. *Read-only*.
+
+    ##Identifier Attributes
+
+    OSF Identifier entities have the "identifiers" `type`.
+
+        name           type                   description
+        ----------------------------------------------------------------------------
+        category       string                 e.g. 'ark', 'doi'
+        value          string                 the identifier value itself
+
+    ##Links
+
+        self: this identifier's detail page
+
+    ##Relationships
+
+    ###Referent
+
+    The identifier is refers to this preprint.
+
+    ##Actions
+
+    *None*.
+
+    ##Query Params
+
+     Identifiers may be filtered by their category.
+
+    #This Request/Response
+
+    """
+
+    permission_classes = (
+        PreprintPublishedOrAdmin,
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    serializer_class = PreprintIdentifierSerializer
+    required_read_scopes = [CoreScopes.IDENTIFIERS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    preprint_lookup_url_kwarg = 'preprint_id'
+
+    view_category = 'identifiers'
+    view_name = 'identifier-list'
+
+    def get_preprint(self, check_object_permissions=True):
+        preprint = get_object_or_error(
+            PreprintService,
+            self.kwargs[self.preprint_lookup_url_kwarg],
+            display_name='preprint'
+        )
+
+        # May raise a permission denied
+        if check_object_permissions:
+            self.check_object_permissions(self.request, preprint)
+        return preprint
+
+    # overrides ODMFilterMixin
+    def get_default_odm_query(self):
+        return MODMQ('pk', 'in', self.get_preprint().identifiers.values_list('pk', flat=True))
```

Comment:
This isn't ideal--this is doing at least 3 queries (`get_preprint`, `.identifiers`, `pk__in`). I'm wondering if we can rid of the outer query by doing:

```python
def get_queryset(self):
    return self.get_preprint().identifiers.all()
```

That said, I see this code is copied from `IdentifierList`, so this is fine for now. I've created a follow-up ticket to look into this: https://openscience.atlassian.net/browse/OSF-7932

No action necessary for now.